### PR TITLE
Code commits related changes of SATQE-3456 and fixed duplicate org and product.

### PR DESCRIPTION
**Pull request by devendra104**
Code commits related changes of SATQE-3456 and fixed duplicate org and product creation

### CODE REVIEW

File: tests/foreman/cli/test_contentview.py [hunk 1]
```diff
@@ -5302,7 +5302,6 @@ def test_positive_arbitrary_file_sync_over_capsule(self):
         :CaseLevel: System
         """
 
-    @stubbed()
     @tier2
     def test_positive_arbitrary_file_repo_promotion(self):
         """Check arbitrary files availability on Environment after Content
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_contentview.py]: The docstring is complete? 'after content' what? do you mean content view promotion?

File: tests/foreman/cli/test_contentview.py [hunk 2]
```diff
@@ -5323,7 +5322,26 @@ def test_positive_arbitrary_file_repo_promotion(self):
         :expectedresults: Check arbitrary files from FR is available on
             environment
 
-        :CaseAutomation: notautomated
+        :CaseAutomation: automated
 
         :CaseLevel: Integration
         """
+        org = ContentViewFileRepoTestCase.org
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_contentview.py]: why not self.org?  Just to make code consistent

File: tests/foreman/cli/test_contentview.py [hunk 3]
```diff
@@ -5323,7 +5322,26 @@ def test_positive_arbitrary_file_repo_promotion(self):
...
+        cv = make_content_view({u'organization-id': org['id']})
+        product_info = ContentViewFileRepoTestCase.product
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_contentview.py]: same as above

File: tests/foreman/cli/test_contentview.py [hunk 4]
```diff
@@ -5323,7 +5322,26 @@ def test_positive_arbitrary_file_repo_promotion(self):
         :expectedresults: Check arbitrary files from FR is available on
             environment
 
-        :CaseAutomation: notautomated
+        :CaseAutomation: automated
 
         :CaseLevel: Integration
         """
+        org = ContentViewFileRepoTestCase.org
+        cv = make_content_view({u'organization-id': org['id']})
```
Comment (vijay8451) [inline tests/foreman/cli/test_contentview.py]:
```suggestion
        cv = make_content_view({u'organization-id': self.org['id']})
```
in above way ..

File: tests/foreman/cli/test_contentview.py [hunk 5]
```diff
@@ -5323,7 +5322,26 @@ def test_positive_arbitrary_file_repo_promotion(self):
...
+        product_info = ContentViewFileRepoTestCase.product
+        repo = self._make_file_repository_upload_contents(
+            {u'content-type': 'file', u'organization-id': org['id'], u'product-id':
+                product_info['id']})
```
Comment (vijay8451) [inline tests/foreman/cli/test_contentview.py]:
```suggestion
                self.product['id']})
```
same here

File: tests/foreman/cli/test_contentview.py (hunk 2 above)
Comment (vijay8451) [inline tests/foreman/cli/test_contentview.py]:
```suggestion

```
Looks not require as self.org is an object and can directly call.

File: tests/foreman/cli/test_contentview.py (hunk 3 above)
Comment (vijay8451) [inline tests/foreman/cli/test_contentview.py]:
```suggestion

```
same here

File: tests/foreman/cli/test_contentview.py (hunk 2 above)
Comment (devendra104) [reply tests/foreman/cli/test_contentview.py]: Agreed, Omkar same has been taken care.

File: tests/foreman/cli/test_contentview.py (hunk 1 above)
Comment (devendra104) [reply tests/foreman/cli/test_contentview.py]: Yes, Here we are validating the arbitrary file availability after content- view version promotion.

File: tests/foreman/cli/test_contentview.py [hunk 6]
```diff
@@ -5323,7 +5322,26 @@ def test_positive_arbitrary_file_repo_promotion(self):
...
+        ContentView.add_repository({u'id': cv['id'], u'repository-id': repo['id'],
```
Comment (oshtaier) [inline tests/foreman/cli/test_contentview.py]: use proper formatting and indentation
```
        ContentView.add_repository({
            'id': cv['id'], 
            'repository-id': repo['id'],
            'organization-id': org['id']
})
```

Please apply to all places

Comment (devendra104) [reply tests/foreman/cli/test_contentview.py]: Thanks, Same has been taken care.

**Comment by devendra104** [comment]
Execution Result after fixed:
(robo_env) [desingh@desingh robottelo]$ pytest -v tests/foreman/cli/test_contentview.py::ContentViewFileRepoTestCase::test_positive_arbitrary_file_repo_promotion
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.8.0, pluggy-0.9.0 -- /home/desingh/robottalo_backup_project/robottelo/robo_env/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/desingh/robottalo_backup_project/robottelo, inifile:
plugins: services-1.3.1, mock-1.10.0, helpers-namespace-2019.1.8, forked-1.0.2, cov-2.7.1
collecting ... 2019-05-06 14:53:08 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                                                  

tests/foreman/cli/test_contentview.py::ContentViewFileRepoTestCase::test_positive_arbitrary_file_repo_promotion PASSED                                                                                      [100%]

============================================================================================ 1 passed in 72.33 seconds ============================================================================================
(robo_env) [desingh@desingh robottelo]$ git status

File: tests/foreman/cli/test_contentview.py (hunk 3 above)
Comment (devendra104) [reply tests/foreman/cli/test_contentview.py]: Same has been incorporated.

File: tests/foreman/cli/test_contentview.py (hunk 2 above)
Comment (devendra104) [reply tests/foreman/cli/test_contentview.py]: Right, Same has been taken care.

File: tests/foreman/cli/test_contentview.py (hunk 5 above)
Comment (devendra104) [reply tests/foreman/cli/test_contentview.py]: Ack.

File: tests/foreman/cli/test_contentview.py (hunk 4 above)
Comment (devendra104) [reply tests/foreman/cli/test_contentview.py]: Ack

File: tests/foreman/cli/test_contentview.py (hunk 3 above)
Comment (devendra104) [reply tests/foreman/cli/test_contentview.py]: Same has been taken care.

File: tests/foreman/cli/test_contentview.py [hunk 7]
```diff
@@ -5323,7 +5322,40 @@ def test_positive_arbitrary_file_repo_promotion(self):
...
+        })
+        repo = self._make_file_repository_upload_contents({
+            u'content-type': 'file',
+            u'organization-id': self.org['id'],
+            u'product-id': self.product['id']
+        })
+        ContentView.add_repository({
+            u'id': cv['id'],
+            u'repository-id': repo['id'],
+            u'organization-id': self.org["id"]})
```
Comment (oshtaier) [inline tests/foreman/cli/test_contentview.py]: use same quotes format and proper indentation
```
            u'organization-id': self.org['id']
})
```

File: tests/foreman/cli/test_contentview.py [hunk 8]
```diff
@@ -5323,7 +5322,40 @@ def test_positive_arbitrary_file_repo_promotion(self):
...
+        env = make_lifecycle_environment({
+            u'organization-id': self.org['id']
+        })
+        ContentView.publish({
```
Comment (oshtaier) [inline tests/foreman/cli/test_contentview.py]: that is fine to be in one line as string has less than 99 symbols:
`ContentView.publish({u'id': cv['id']})`

File: tests/foreman/cli/test_contentview.py [hunk 9]
```diff
@@ -5323,7 +5322,40 @@ def test_positive_arbitrary_file_repo_promotion(self):
...
+        })
+        content_view_info = ContentView.version_info({
+            u'content-view-id': cv['id'],
+            u'version': 1
+        })
+        ContentView.version_promote({
+                u'id': content_view_info['id'],
+                u'to-lifecycle-environment-id': env['id']
+        })
+        expected_repo = ContentView.version_info({
```
Comment (oshtaier) [inline tests/foreman/cli/test_contentview.py]: We have:
```
        :expectedresults: Check arbitrary files from FR is available on
            environment
```
so it is more available in content view version?

**Comment by devendra104** [comment]
Test execution result after Fixed:

```
(robo_env) [desingh@desingh robottelo]$ pytest -v tests/foreman/cli/test_contentview.py::ContentViewFileRepoTestCase::test_positive_arbitrary_file_repo_promotion
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.8.0, pluggy-0.9.0 -- /home/desingh/robottalo_backup_project/robottelo/robo_env/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/desingh/robottalo_backup_project/robottelo, inifile:
plugins: services-1.3.1, mock-1.10.0, helpers-namespace-2019.1.8, forked-1.0.2, cov-2.7.1
collecting 1 item                                                                                                                                                                                                 2019-05-06 19:10:23 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                                                  

tests/foreman/cli/test_contentview.py::ContentViewFileRepoTestCase::test_positive_arbitrary_file_repo_promotion PASSED
```

File: tests/foreman/cli/test_contentview.py [hunk 10]
```diff
@@ -5338,13 +5338,12 @@ def test_positive_arbitrary_file_repo_promotion(self):
         ContentView.add_repository({
             u'id': cv['id'],
             u'repository-id': repo['id'],
-            u'organization-id': self.org["id"]})
+            u'organization-id': self.org["id"]
```
Comment (oshtaier) [inline tests/foreman/cli/test_contentview.py]: I mentioned single quotes in my previous comments
`self.org['id']`

File: tests/foreman/cli/test_contentview.py [hunk 11]
```diff
@@ -5320,7 +5320,7 @@ def test_positive_arbitrary_file_repo_promotion(self):
             1. Promote the CV to the Environment
 
         :expectedresults: Check arbitrary files from FR is available on
-            environment
+            content view version environment
```
Comment (oshtaier) [inline tests/foreman/cli/test_contentview.py]: I believe
`content view version` without `environment`

Comment (devendra104) [reply tests/foreman/cli/test_contentview.py]: As per the test step, we have to check the repository avaiability in published content-view's particular version having specific environment.
So I think it is correct.

**Comment by devendra104** [comment]
Test Result after fixed:
```
(robo_env) [desingh@desingh robottelo]$ pytest -v tests/foreman/cli/test_contentview.py::ContentViewFileRepoTestCase::test_positive_arbitrary_file_repo_promotion
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.8.0, pluggy-0.9.0 -- /
collected 1 item                                                                                                                                                                                                  

tests/foreman/cli/test_contentview.py::ContentViewFileRepoTestCase::test_positive_arbitrary_file_repo_promotion PASSED                                                                                      [100%]

=========================================================================================== 1 passed in 108.21 seconds ============================================================================================
```

**Comment by devendra104** [comment]
> yeah, you understand the idea and logic very good. That's nice =)

Thanks @oshtaier for your valuable feedback.

**Comment by oshtaier** [comment]
Thanks

**Pull request merged**